tabular/src/autogluon/tabular/learner/default_learner.py

```diff
@@ -364,6 +341,59 @@ def set_predefined_weights(self, X, y):
         X[self.sample_weight] = w  # TODO: consider not bundling sample weights inside X
         return X
 
+    def _check_for_non_finite_values(self, X: DataFrame, name: str = "", is_train: bool = False) -> DataFrame:
+        if is_train or (X is not None and self.label in X.columns):
+            X = copy.deepcopy(X)
+
+            # treat None, NaN, INF, NINF as NA
+            X[self.label] = X[self.label].replace([np.inf, -np.inf], np.nan)
+            invalid_labels = X[self.label].isna()
+            if invalid_labels.any():
+                first_invalid_label_idx = invalid_labels.idxmax()
+                raise ValueError(
+                    f"{name} dataset label column cannot contain non-finite values (NaN, Inf, Ninf). First invalid label at data idx: {first_invalid_label_idx}"
+                )
+
+        return X
+
+    def _apply_cleaner_transform(
+        self, X: DataFrame, y_uncleaned: Series, holdout_frac: float | int, holdout_frac_og: float | int, name: str
+    ) -> tuple[DataFrame, Series, Series | None, float | int]:
+        if X is not None and self.label in X.columns:
+            y_og = X[self.label]
+            X = self.cleaner.transform(X)
```

Comment:
Add `is_test` argument and the following:

```suggestion
            y_og = X[self.label]
            len_og = len(X) if is_test else None
            X = self.cleaner.transform(X)
            if is_test and len(X) != len_og:
                raise AssertionError(
                    f"{name} cannot have low frequency classes! Please create a GitHub issue if you see this message, as it should never occur."
                )
```